### PR TITLE
[New] Add tiled layer as basemap

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Add tiled layer as basemap` in `Edit and Manage Data` category.
URL to README: [Add tiled layer as basemap](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-AddTiledLayerAsBasemap/Shared/Samples/Add%20tiled%20layer%20as%20basemap)

## Linked Issue(s)

- `swift/issues/6215`

## How To Test

- Open the sample to see the basemap tiled layer.

## Screenshots

add-tiled-layer-as-basemap

### CODE REVIEW

**Comment by CalebRas** [comment]
The [iOS implementation](https://github.com/Esri/arcgis-runtime-samples-ios/tree/main/arcgis-ios-sdk-samples/Layers/ArcGIS%20tiled%20layer%20(tile%20cache)) has a button to view the other tile package files in the bundle and documents. This was not part of the design, so I did not implement this. Let me know if you think it is needed.

**Review by njarecha — DISMISSED** [review]
Looks good

File: Shared/Samples/Add tiled layer as basemap/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## How to use the sample
+
+Launch the app to view the "San Francisco offline tile package" as the basemap.
+
+## How it works
+
+1. Create a `TileCache`, specifying the path to the local tile package.
+2. Create an `ArcGISTiledLayer` with the tile cache.
+3. Create a `Basemap` with the tiled layer.
+4. Create an `ArcGISMap` with the basemap and pass it to a MapView.
```
Comment (pgruenler) [inline Shared/Samples/Add tiled layer as basemap/README.md:20]:
```suggestion
4. Create a `Map` with the basemap and pass it to a MapView.
```

**Review by pgruenler — COMMENTED** [review]
Looks good, just one suggestion.

**Review by pgruenler — APPROVED** [review]

**Review by njarecha — APPROVED** [review]

**Comment by CalebRas** [comment]
Thank you both!

**Pull request merged**